### DISCUSSION TITLE
Processes

### DISCUSSION THREAD

**Discussion by azinit** [feature-sliced/documentation#386] [DocDraft]
Мифический слой, с которым крайне мало кто работал
(т.к. нужен он далеко не во всех проектах)

В последнее время оч часто стали спрашивать
https://t.me/feature_sliced/9175

**Comment by azinit** [comment]
https://t.me/feature_sliced/9782
https://t.me/feature_sliced/9717
https://t.me/feature_sliced/9282
https://t.me/feature_sliced/8528
https://t.me/feature_sliced/8437
https://t.me/feature_sliced/7601
https://t.me/feature_sliced/7082
https://t.me/feature_sliced/6179
https://t.me/feature_sliced/5798
https://t.me/feature_sliced/3677

**Reply by dovranJorayev** [reply]
Как пример для логики которую я описываю в слое проецессов

1. Логика защищенного роутинга который будет проверять токен, профиль, права доступа и т.д. и разрешать или запрещать доступ к страницам
3. Для логики для редиректа на логин при 401 статусе ошибки любога запроса в приложении
4. Если приложение загружается из конфига чтобы не пихать эту логику в app ложу в слой процессов. Если у нас монорепозитория и  нужен переиспользовать эту логику то трудно ее извлекать из app
6. Кейсы где нужно развернуть соединение с web socket 
7. Автоматическая отправка аналитики при переходе по страницам
8. Отправка аналитики по текущему устройству  при входе в приложение (возможно сделать фичей, но очень наглядно видеть такое в процессах а не в app)

Некоторые пункты выше можно раскидать по страницам и фичам. Слой processes дает определенную гибкость при работе с кейсами выше и с другими похожими кейсами сквозной логики, обычно лежащей выше слоя pages, но не вписывающейся в app так как там должна быть в идеале только оркестрация и инициализация приложения
